core/backend/src/ChannelControl.ts

```diff
@@ -86,27 +67,20 @@ export class ChannelAdmin implements ChannelControl {
     this._allowedChannels.delete(channelKey);
     this._allowedModels.clear();
   }
-  public get hasChannels(): boolean {
-    if (undefined === this._hasChannels) {
-      try {
-        this._hasChannels = this._iModel.withStatement(`SELECT 1 FROM ${ChannelAdmin.channelClassName}`, (stmt) => stmt.step() === DbResult.BE_SQLITE_ROW, false);
-      } catch (e) {
-        // iModel doesn't have channel class in its BIS schema
-        this._hasChannels = false;
-      }
-    }
-    return this._hasChannels;
-  }
   public getChannelKey(elementId: Id64String): ChannelKey {
-    if (!this.hasChannels || elementId === IModel.rootSubjectId)
+    if (elementId === IModel.rootSubjectId)
       return ChannelControl.sharedChannelName;
 
-    const channel = this._iModel.withPreparedStatement(`SELECT Owner FROM ${ChannelAdmin.channelClassName} WHERE Element.Id=?`, (stmt) => {
-      stmt.bindId(1, elementId);
-      return DbResult.BE_SQLITE_ROW === stmt.step() ? stmt.getValue(0).getString() : undefined;
-    });
-    if (channel !== undefined)
-      return channel;
+    try {
+      const channel = this._iModel.withPreparedStatement(`SELECT Owner FROM ${ChannelAdmin.channelClassName} WHERE Element.Id=?`, (stmt) => {
+        stmt.bindId(1, elementId);
+        return DbResult.BE_SQLITE_ROW === stmt.step() ? stmt.getValue(0).getString() : undefined;
+      });
+      if (channel !== undefined)
+        return channel;
+    } catch {
+      return ChannelControl.sharedChannelName;
```

Comment:
add a comment that this happens if the db is too old and the channel root aspect class isn't present. In that case we just say everything is in the shared channel.

Reply:
I don't know how I missed adding that comment. Sorry!